lib/danger/dangerfile.rb

```diff
@@ -21,9 +21,32 @@ def to_s
       'Dangerfile'
     end
 
+    # Iterates through the DSL's attributes, and table's the output
+    def print_known_info
+      puts "Danger v#{Danger::VERSION}"
+      width = AvailableValues.all.map(&:to_s).map(&:length).max
+      puts "DSL Attributes:"
+      puts "-" * (width + 4)
+      AvailableValues.all.each do |value|
+        spaces = (width - value.to_s.length)
+        puts "| #{value.to_s.blue}#{' ' * spaces} | #{self.send(value)}"
+      end
+      puts "-" * (width + 4)
+
+      puts "Metadata:"
+      puts "#{'SCM'.blue}      : #{env.scm.class}"
+      puts "#{'Source'.blue}   : #{env.ci_source.class}"
+      puts "           #{'Base commit'.blue} : #{env.ci_source.base_commit}"
+      puts "           #{'HEAD commit'.blue} : #{env.ci_source.head_commit}"
+      puts "           git diff  #{env.ci_source.base_commit} #{env.ci_source.head_commit}".yellow
+      puts "#{'Requests'.blue} : #{env.request_source.class}"
+    end
```

Comment:
It's a start
![screen shot 2016-02-08 at 10 43 56](https://cloud.githubusercontent.com/assets/49038/12883599/deb2b90e-ce50-11e5-8e49-2181a642e1c6.png)